packages/site/cms-site/src/sitePreview/SitePreviewUtils.ts

```diff
@@ -64,7 +65,10 @@ export async function sitePreviewRoute(request: NextRequest, graphQLFetch: Graph
 
     draftMode().enable();
 
-    return redirect(params.get("path") || "/");
+    const apiRoutePostfix = options?.apiRoutePostfix ?? "/api/site-preview";
+    const basePath = request.nextUrl.pathname.split(apiRoutePostfix)[0];
```

Comment:
IMO this would be easier to understand: `const basePath = request.nextUrl.pathname.replace(apiRoutePostfix, "");`

Reply:
3ceb8784b14d9bf9f55b25170b54329895d98b65